platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/addressbook/RandomRosterBuilder.java

```diff
@@ -0,0 +1,311 @@
+     */
+    @NonNull
+    public RandomRosterBuilder withSize(final int size) {
+        this.size = size;
+        return this;
+    }
+
+    /**
+     * Set the average weight for an address. If the weight distribution strategy is
+     * {@link WeightDistributionStrategy#BALANCED}, all addresses will have this weight. If the weight distribution
+     * strategy is {@link WeightDistributionStrategy#GAUSSIAN}, this will be the mean weight.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withAverageWeight(final long averageWeight) {
+        this.averageWeight = averageWeight;
+        return this;
+    }
+
+    /**
+     * Set the standard deviation for the weight for an address. Ignored unless the weight distribution strategy is
+     * {@link WeightDistributionStrategy#GAUSSIAN}.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withWeightStandardDeviation(final long weightStandardDeviation) {
+        this.weightStandardDeviation = weightStandardDeviation;
+        return this;
+    }
+
+    /**
+     * Set the minimum weight for an address. Overrides the weight generation strategy.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withMinimumWeight(final long minimumWeight) {
+        this.minimumWeight = minimumWeight;
+        return this;
+    }
+
+    /**
+     * Set the maximum weight for an address. Overrides the weight generation strategy.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withMaximumWeight(final long maximumWeight) {
+        this.maximumWeight = maximumWeight;
+        return this;
+    }
+
+    /**
+     * Set the strategy used for deciding distribution of weight.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withWeightDistributionStrategy(
+            final WeightDistributionStrategy weightDistributionStrategy) {
+
+        this.weightDistributionStrategy = weightDistributionStrategy;
+        return this;
+    }
+
+    /**
+     * Specify if real cryptographic keys should be generated (default false). Warning: generating real keys is very
+     * time consuming.
+     *
+     * @param realKeysEnabled if true then generate real cryptographic keys
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withRealKeysEnabled(final boolean realKeysEnabled) {
+        this.realKeys = realKeysEnabled;
+        return this;
+    }
+
+    /**
+     * Get the private keys for a node. Should only be called after the roster has been built and only if
+     * {@link #withRealKeysEnabled(boolean)} was set to true.
+     *
+     * @param nodeId the node id
+     * @return the private keys
+     * @throws IllegalStateException if real keys are not being generated or the roster has not been built
+     */
+    @NonNull
+    public KeysAndCerts getPrivateKeys(final NodeId nodeId) {
```

Comment:
Should we `@NonNull` the nodeId parameter?

Reply:
Added @NonNull.